### ISSUE TITLE
Loss weighting from DALLE paper

### ISSUE THREAD

**Issue by kobiso** [lucidrains/DALLE-pytorch#133]
In DALLE paper, they applied separated weights for text and image cross-entropy losses.

>We normalize the cross-entropy losses for the text and image tokens by the total number of each kind in a batch of data. Since we are primarily interested in image modeling, we multiply the cross-entropy loss for the text by 1/8 and the cross-entropy loss for the image by 7/8. The objective is optimized using Adam with exponentially weighted iterate averaging.

So, I did following experiments and found some small improvements. (I'm not sure cuz I can only do quantitative evaluation 😅)
I will PR this :)

### Training graph
- **Experimental details**: https://github.com/lucidrains/DALLE-pytorch/discussions/131
- **Models**
  - Green: w/o loss_weight
  - Pink: w/ loss_weight

![image](https://user-images.githubusercontent.com/12422441/112794197-7d39b380-90a1-11eb-8c3a-90de08f927d1.png)

### Visualization
![image](https://user-images.githubusercontent.com/12422441/112794520-f2a58400-90a1-11eb-92d3-abda5629bb05.png)
![image](https://user-images.githubusercontent.com/12422441/112794543-ffc27300-90a1-11eb-83a6-00c07316b395.png)

![image](https://user-images.githubusercontent.com/12422441/112794531-f89b6500-90a1-11eb-984d-99018da54195.png)
![image](https://user-images.githubusercontent.com/12422441/112794561-0650ea80-90a2-11eb-8b04-6aafbbaa0c7b.png)
